SL_ArabicOCR.py

```diff
@@ -102,10 +101,9 @@ def show_pdf(file_path):
             else:
                 num_of_pages = 1
             print(num_of_pages)
-            with pdfium.PdfContext(pdf_input) as pdf:
-                for page in range(num_of_pages):
-                    image = pdfium.render_page_topil(pdf, page)
-                    images.append(image)
+            with pdfium.PdfDocument(pdf_input) as pdf:
```

Comment:
Can you clarify what kind of object `pdf_input` is? It looks like this may be some special streamlit thing.
While pypdfium2 only takes file paths, byte data, or byte buffers, I believe this worked and will continue to work because pypdfium2 detected your object as a byte buffer using `hasattr()` checks.
However, maybe this could be implemented in a more explicit way or at least be clarified by comment.

Reply:
I've decided to remove implicit reading of byte buffers in the multi-page renderer, so the PR does not work as-is anymore.
Your current code in main, however, should continue to work because it's using the single-page renderer.

Reply:
I've fixed up the PR to do an explicit `read()`